backend/fms_core/viewsets/sample_pooled.py

```diff
@@ -0,0 +1,117 @@
+from rest_framework.decorators import action
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.pagination import LimitOffsetPagination
+from rest_framework.exceptions import APIException
+from fms_core.models import Biosample, DerivedSample, DerivedBySample, Index, Library, LibraryType, Project, Sample, SampleKind
+from fms_core.filters import PooledSampleFilter
+
+# Custom serializers
+# These serializers serialize the data we want to receive in pool samples.
+
+EXCLUDE_BOOKKEEPING = ['created_at', 'updated_at', 'created_by', 'updated_by', 'deleted']
+
+class IndexSerializer(serializers.ModelSerializer):
+        index_set = serializers.CharField(read_only=True, source="index_set.name")
+        index_structure = serializers.CharField(read_only=True, source="index_structure.name")
+        class Meta:
+            model = Index
+            exclude = EXCLUDE_BOOKKEEPING
+
+class SampleKindSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = SampleKind
+        fields = ['name', 'is_extracted']
+
+class BiosampleSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Biosample
+        exclude = EXCLUDE_BOOKKEEPING
+
+class LibrarySerializer(serializers.ModelSerializer):
+    library_type = serializers.CharField(read_only=True, source='library_type.name')
+    platform = serializers.CharField(read_only=True, source='platform.name')
+    index = IndexSerializer(read_only=True, required=False)
+    class Meta:
+        model = Library
+        exclude = EXCLUDE_BOOKKEEPING
+        depth = 1
+
+class ProjectSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Project
+        fields = ['name', 'id']
+
+class DerivedSampleSerializer(serializers.ModelSerializer):
+    library = LibrarySerializer(required=False)
+    biosample = BiosampleSerializer(required=False)
+    tissue_source = SampleKindSerializer(required=False)
+    sample_kind=SampleKindSerializer(required=False)
+    project = ProjectSerializer(required=False)
+    
+    class Meta:
+        model = DerivedSample
+        exclude = EXCLUDE_BOOKKEEPING
+        depth = 1
+        
+
+class PooledSampleSerializer(serializers.Serializer):
+    ''' Serializes a DerivedBySample object, representing a pooled sample. 
+        The result is a nested data structure containing all of the information about the
+        pooled sample needed by the frontend for display.
+    '''
+    volume_ratio = serializers.DecimalField(max_digits=20, decimal_places=3, read_only=True)
+    derived_sample = DerivedSampleSerializer()
+    # Since DerivedBySample doesn't have its own id field, we use the derived_sample id
+    # as a top level id in the returned data structure. The UX needs this for 'objectsById' stuff.
+    id = serializers.IntegerField(read_only = True, source='derived_sample.id')
+    parent_sample_name=serializers.SerializerMethodField()
```

Comment:
you can remove those parent infos if you don't use them.

Reply:
Keeping them for now, will be replaced with a better recursive lookup in a separate ticket